ion/binarywriter.go

```diff
@@ -495,10 +495,16 @@ func (w *binaryWriter) beginValue(api string) error {
 		ids := make([]uint64, len(as))
 		idlen := uint64(0)
 
+		var id uint64
+		var err error
 		for i, a := range as {
-			id, err := w.resolve(api, a)
-			if err != nil {
-				return err
+			if a.Text != nil {
+				id, err = w.resolve(api, *a.Text)
+				if err != nil {
+					return err
+				}
+			} else {
+				id = 0
```

Comment:
Is this intended to support pass-through for Ion data containing unresolveable symbols?

I believe this also allows a user to write:
```go
writer.Annotation(nil)
```
and produce a `$0` in the output. That should be an error instead.

Down the road, when we offer a `AnnotationSymbolToken(SymbolToken)` method alongside the `Annotation(string)` method, we would want this to use the `LocalSID` field's value if it was set (i.e. `!= -1`) instead of using `0`.

Reply:
id is now set to the `LocalSID` if it's not -1 (and returns an error is it is -1).